src/main/java/pl/edu/mimuw/queue/AbstractIntQueue.java

```diff
@@ -1,39 +1,149 @@
 package pl.edu.mimuw.queue;
 
 public abstract class AbstractIntQueue {
-
-  // TODO: you can make changes with this class fields, constructors
-  //  but also add some methods but the specified methods cannot be changed (you
-  //  can change them not to be abstract and provide some implementation for them,
-  //  but they have to have the same names, arguments and returned values)
-
-  /**
-   * Adds element to the queue.
-   *
-   * @throws NullPointerException if the specified element is null
-   */
-  public abstract void offer(Integer x);
-
-  /**
-   * @return the head of this queue, or {@code null} if this queue is empty
-   * and don't remove the element from the queue
-   */
-  public abstract Integer peek();
-
-  /**
-   * @return the head of this queue, or {@code null} if this queue is empty
-   * and remove the element from the queue
-   */
-  public abstract Integer poll();
-
-  /**
-   * @return readable representation of ordered queue elements
-   */
-  @Override
-  public abstract String toString();
-
-  /**
-   * @return the number of elements in this queue
-   */
-  public abstract int size();
+	private int size;
+	private IntQueueNode left, right;
```

Comment:
jeżeli całą implementację tu wrzucasz, to nie udostępniaj jej całej publicznie - w takim podejściu u Ciebie LIFO i FIFO stały się deque (bo mają publicznie dostępne metody do modyfikacji początku i końca) a zdecydowanie nie o to nam chodziło - wystarczy zrobić je jako protected

Ponadto i tak nie wszystkie metody muszą być w klasie abstrakcyjnej